pybamm/parameters/lithium_ion_parameters.py

```diff
@@ -663,16 +745,32 @@ def D_n(self, c_s_n, T):
         "Dimensionless negative particle diffusivity"
         sto = c_s_n
         T_dim = self.Delta_T * T + self.T_ref
-        return self.D_n_dimensional(sto, T_dim) / self.D_n_dimensional(
-            pybamm.Scalar(1), self.T_ref
+        if self.options["particle cracking"] is not None:
+            mech_effects = (
+                1 + self.theta_n * sto / T_dim * self.T_ref * self.flag_mechanics
+            )
+        else:
+            mech_effects = 1
+        return (
+            self.D_n_dimensional(sto, T_dim)
+            / self.D_n_dimensional(pybamm.Scalar(1), self.T_ref)
+            * mech_effects
         )
 
     def D_p(self, c_s_p, T):
         "Dimensionless positive particle diffusivity"
         sto = c_s_p
         T_dim = self.Delta_T * T + self.T_ref
-        return self.D_p_dimensional(sto, T_dim) / self.D_p_dimensional(
-            pybamm.Scalar(1), self.T_ref
+        if self.options["particle cracking"] is not None:
+            mech_effects = (
+                1 + self.theta_p * sto / T_dim * self.T_ref * self.flag_mechanics
```

Comment:
in the reference the mechanical effects are given by `1+theta*c_bar` with `c_bar = c-c_ref`

should `sto` on line 766 be `sto*c_max-c_0`?

Reply:
as written the units are wrong (`sto=c/c_max` so is dimensionless)

Reply:
I think there are some inconsistencies with the way things have been non-dimensionalised

e.g. I think 

```python3
        self.theta_p = self.theta_p_dim * self.c_p_max / self.Delta_T
        self.theta_n = self.theta_n_dim * self.c_n_max / self.Delta_T
```
and 
```python3
    def D_n(self, c_s_n, T):
        "Dimensionless negative particle diffusivity"
        sto = c_s_n
        T_dim = self.Delta_T * T + self.T_ref
        if self.options["particle cracking"] is not None:
            mech_effects = 1 + self.theta_n * (sto - self.c_n_0 / self.c_n_max) / (
                T + 1 / self.Theta
            )
        else:
            mech_effects = 1
        return (
            self.D_n_dimensional(sto, T_dim)
            / self.D_n_dimensional(pybamm.Scalar(1), self.T_ref)
            * mech_effects
        )

    def D_p(self, c_s_p, T):
        "Dimensionless positive particle diffusivity"
        sto = c_s_p
        T_dim = self.Delta_T * T + self.T_ref
        if self.options["particle cracking"] is not None:
            mech_effects = 1 + self.theta_p * (sto - self.c_p_0 / self.c_p_max) / (
                T + 1 / self.Theta
            )
        else:
            mech_effects = 1
        return (
            self.D_p_dimensional(sto, T_dim)
            / self.D_p_dimensional(pybamm.Scalar(1), self.T_ref)
            * mech_effects
        )
```

I think what would be a better solution in the long term would be to make `theta_M` a variable, and then change the relevant code in the particle models to have the `(1+theta_M)`. for "no cracking" models `theta_M=0`.

it's probably ok to manipulate the parameter for now, but would it make more sense to change the dimensional diffusion coefficient to account for mechanics? now it is a bit inconsistent because the dimensionless `D` accounts for the mechanics, but the dimensional one doesn't. having had a quick play it looks like the scalings might be the cause of the issue with it not matching the data

Reply:
yeah I think better to write as

```python3
    def D_n_dimensional(self, sto, T):
        """Dimensional diffusivity in negative particle. Note this is defined as a
        function of stochiometry"""
        inputs = {"Negative particle stoichiometry": sto, "Temperature [K]": T}
        if self.options["particle cracking"] is not None:
            mech_effects = 1 + self.theta_n_dim * (sto * self.c_n_max - self.c_n_0) / T
        else:
            mech_effects = 1
        return (
            pybamm.FunctionParameter("Negative electrode diffusivity [m2.s-1]", inputs)
            * mech_effects
        )

    def D_p_dimensional(self, sto, T):
        """Dimensional diffusivity in positive particle. Note this is defined as a
        function of stochiometry"""
        inputs = {"Positive particle stoichiometry": sto, "Temperature [K]": T}
        if self.options["particle cracking"] is not None:
            mech_effects = 1 + self.theta_p_dim * (sto * self.c_p_max - self.c_p_0) / T
        else:
            mech_effects = 1
        return (
            pybamm.FunctionParameter("Positive electrode diffusivity [m2.s-1]", inputs)
            * mech_effects
        )
```
and 
```python3
    def D_n(self, c_s_n, T):
        "Dimensionless negative particle diffusivity"
        sto = c_s_n
        T_dim = self.Delta_T * T + self.T_ref
        return self.D_n_dimensional(sto, T_dim) / self.D_n_dimensional(
            pybamm.Scalar(1), self.T_ref
        )

    def D_p(self, c_s_p, T):
        "Dimensionless positive particle diffusivity"
        sto = c_s_p
        T_dim = self.Delta_T * T + self.T_ref
        return self.D_p_dimensional(sto, T_dim) / self.D_p_dimensional(
            pybamm.Scalar(1), self.T_ref
        )
```

though this doesn't change the results

Reply:
Agree and thanks for this suggestion.